pkg/provisioner/ironic/ironic.go

```diff
@@ -443,6 +445,35 @@ func (p *ironicProvisioner) ValidateManagementAccess(data provisioner.Management
 
 		updater.SetTopLevelOpt("name", ironicNodeName(p.objectMeta), ironicNode.Name)
 
+		var bmcAddressChanged bool
+		var newAddressKey string
+		var newAddressValue interface{}
+		var ironicAddressKey string
+		var ironicAddressValue interface{}
+		for key, value := range driverInfo {
+			if found, _ := regexp.MatchString("address", key); found {
```

Comment:
This is not very robust. At the very least, I'd use something like "_address$". Ideally, let the bmcAccess figure that out.

Reply:
I've changed the regex and tried to reduce the number of variables needed. Does it look better?